Source/WebKit/NetworkProcess/storage/CacheStorageDiskStore.cpp

```diff
@@ -318,12 +317,14 @@ std::optional<CacheStorageRecord> CacheStorageDiskStore::readRecordFromFileData(
         if (storedInfo->metaData.bodyHash != computeSHA1(bodyData, m_salt))
             return std::nullopt;
 
+        // FIXME: avoid copying inline body data here, perhaps by adding offset support to
+        // MappedFileData, or by taking a read-only virtual copy of bodyData.
         responseBody = WebCore::SharedBuffer::create(bodyData.data(), bodyData.size());
     } else {
-        if (blobBuffer.isEmpty())
+        if (!blobBuffer)
             return std::nullopt;
 
-        auto sharedBuffer = WebCore::SharedBuffer::create(blobBuffer.data(), blobBuffer.size());
+        auto sharedBuffer = WebCore::SharedBuffer::create(WTFMove(blobBuffer));
```

Comment:
You would think that since SharedBuffer has a constructor that takes a `MappedFileData`, then when you send such a SharedBuffer over IPC that it doesn't require a memcpy. Well, that isn't true (see `ArgumentCoder<WebCore::FragmentedSharedBuffer>::encode`), but fixing that probably doesn't belong in this patch.